Kanban/QSFU_kanban.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+#==============================================================================
+
+from redminelib import Redmine
+import pdb
+import datetime
+import logging
+import json
+import statistics
+import configparser
+
+config = configparser.ConfigParser()
+config.read('config_kanban.ini')
+
+URL = config['API']['URL']
+KEY = config['API']['ApiKey']
+
+redmine = Redmine(URL, key=KEY)
+
+prjlist = ['suseqa', 'openqav3', 'openqatests']
+taglist = ['y', 'epic', 'saga', 'fate']
+cycleList = []
+leadList = []
+
+id_codes = {
+        'New' : '1',
+        'In Progress' : '2',
+        'Resolved' : '3',
+        'Feedback' : '4',
+        'Workable' : 12
+        }
+
+
+def went_in_progress(jsn):
+    if jsn.get("name", "invalid_default") == "status_id":
+        if jsn["new_value"] == id_codes.get("In Progress"):
+            return jsn["old_value"] in [id_codes["Workable"], id_codes["New"]]
+    return False
+
+def is_story(tckt):
+    if any(s in tckt.subject for s in taglist):
+        return False
+    return True
+
+for prj in prjlist:
+    tickets = redmine.issue.filter(project_id=prj, status_id=id_codes["Resolved"])
+    u_tickets = [t for t in tickets if '[u]' in t.subject]
+
+    for tckt in u_tickets:
+        # only consider normal [u] tickets
+        if not is_story(tckt):
+            continue
+
+        #try:
+        #    print(t.subject)
+        #except:
+        #    print("Could not print subject")
+
+        crtDate = tckt.created_on
+        endDate = tckt.closed_on
+        # set an invalid default
+        prgDate = '-1'
+
+        #print("Creation date: " + str(crtDate))
+        #print("Close date: " + str(endDate))
+
+        journals = tckt.journals
+        for jrn in journals:
+            try:
+                # look at first entry to not get an array
+                jDetails = json.loads(json.dumps(jrn.details[0]))
```

Comment:
I'd rather catch this exception at the level where "not an array" is expected (I guess in `went_in_progress`).
The way you did it, is checking for something simple ("not an array") by executing a parser twice in a row. IMHO a culprit for performance.

Reply:
The mechanism to avoid getting an array type is the [0] at the end, in my understanding the json.dumps is needed to get usable data for the json.loads, that was the only way I got it working in pdb when trying it out, maybe I overlooked sth?